### ISSUE TITLE
REST assured logging config not being respected

### ISSUE THREAD

**Issue by pnsantos** [serenity-bdd/serenity-core#228]
Hi,

I posted this in google groups but maybe here is more appropriate... When using rest-assured standalone (without the serenity plugin) I can configure it to only log when an assertion or the request itself fails. However using the exact same config if I use rest-assured via the serenity plugin all response contents are written to the log.

Is there any way to configure it so that it respects the rest-assured config? I'm using the following set up to configure rest assured:
```
static {
    RestAssured.config = config()
        .redirect(redirectConfig().followRedirects(false))
        .httpClient(httpClientConfig()
                .reuseHttpClientInstance()
                .setParam("http.conn-manager.timeout", 30000L)
                .setParam("http.connection.timeout", 30000)
                .setParam("http.socket.timeout", 30000))
       .logConfig(logConfig().enableLoggingOfRequestAndResponseIfValidationFails());
}
```

**Comment by YamStranger** [comment]
Hello @pnsantos. 

How do you use it with serenity, please provide some example, may be using one of our demo projects  (serenty-demos. you can find here examples of using rest). 

P.S. SerenityRest.rest().config()... - Is not supported now. 

Thanks a lot!

**Comment by pnsantos** [comment]
Hi @YamStranger 

I'm using Serenity with JBehave, they way I structured the project is that I have a class that extends `net.serenitybdd.jbehave.SerenityStory` and then one class for each `.story` file that extends the first class I created.

In that "base" Story class I perform the mentioned initialization. Then in test steps when making HTTP calls I just do something like:

```
rest().given()
        .filter(signer)
        .baseUri(...)
        .basePath(...)
        .queryParam(...)
    .when()
        .get()
    .then()
        .statusCode(....);
```

I'd like to avoid having to configure logging in every single request.... When using this setup with "plain" rest assured (not calling `SerenityRest.rest()` but calling `given()` directly) it works as expected (only when a `then()` assertion fails will the request content get logged) however I've tried a number of ways but can't seem to get Serenity to omit successful API calls.

Tried your suggestion but no luck, request body is still logged for every request.

**Comment by YamStranger** [comment]
Thanks @pnsantos, not it much clear, one more question - under "log" do you mean log that printed to console, during tests executing, or Serenity logs (aggregation reports)?

**Comment by pnsantos** [comment]
Logs to console printed during test execution

**Comment by YamStranger** [comment]
depends on #355

**Comment by YamStranger** [comment]
Hello @pnsantos 
You issue is fixed and can be checked in serenity-core 1.1.31. 

Now same configuration for rest-assurest should word for serenity-rest-assured. 
Also serenity-rest updated and now you can use code same as for rest-assured (started from given() or expect() and so on)

please close this ticket

Thanks a lot!
